mlflow/models/evaluation/base.py

```diff
@@ -653,40 +642,64 @@ def _evaluate(
 
 @experimental
 def evaluate(
-    *,
     model: Union[str, "mlflow.pyfunc.PyFuncModel"],
+    data,
+    *,
+    targets,
     model_type: str,
-    dataset: "mlflow.models.evaluation.EvaluationDataset",
-    run_id=None,
+    dataset_name=None,
+    dataset_path=None,
+    feature_names: list = None,
     evaluators=None,
     evaluator_config=None,
-) -> "mlflow.models.evaluation.EvaluationResult":
+):
     """
     Evaluate a PyFunc model on the specified dataset using one or more specified evaluators, and
     log resulting metrics & artifacts to MLflow Tracking.
 
     :param model: A pyfunc model instance, or a URI referring to such a model.
 
+    :param data: One of the following: A numpy array or list of evaluation features, excluding
+                 labels. Or a Pandas DataFrame, or a spark DataFrame, containing evaluation
+                 features and labels. If `feature_names` argument not specified, all columns are
+                 regarded as feature columns, otherwise column names which match `feature_names`
+                 are regarded as feature columns.
```

Comment:
```suggestion
    :param data: One of the following: 
    
                 - A numpy array or list of evaluation features, excluding labels.
                 
                 - A Pandas DataFrame or Spark DataFrame, containing evaluation features and
                    labels. If `feature_names` argument not specified, all columns are regarded
                    as feature columns. Otherwise, only column names present in `feature_names`
                    are regarded as feature columns.
```